### PR TITLE
feat: Implement reporting of test runs to Sauce Labs Insights

### PR DESCRIPTION
**Pull request by tianfeng92**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

This PR updates the `wdio-sauce-service` package to report test run details to the Insights API upon test completion. This enhancement is particularly beneficial for conducting thorough failure analysis.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-sauce-service/src/api.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                password: opts.accessKey,
+            },
+            baseURL: apiURLMap.get(opts.region),
+        })
+    }
+
+    async create(testRuns: TestRunRequestBody[]) {
+        try {
+            log.debug('Submitting test run to test-runs api', testRuns)
+            await this.api.post<void>('/test-runs/v1/', {
```
Comment (christian-bromann) [inline packages/wdio-sauce-service/src/api.ts:27]: This should be part of the `saucelabs` NPM package which already has a lot of primitives to expose API requests in a simple interface.

Comment (tianfeng92) [reply packages/wdio-sauce-service/src/api.ts:27]: Good call! I'll move it out.

**Review by christian-bromann — APPROVED** [review]
Looks great 👍  thanks a lot!

**Comment by christian-bromann** [comment]
Congratulations on your first contribution to WebdriverIO 🎉


This project thrives on the invaluable involvement of our community, and we are
truly grateful for your contribution. We eagerly anticipate witnessing more of
your exceptional work, so please don't hesitate to inform us if we can assist
you in identifying intriguing areas where you can make further contributions.


Join our lively [Discord](https://discord.webdriver.io/) channel and reach out to us;
we would be delighted to connect with you. Your efforts are deeply appreciated,
and we extend our heartfelt gratitude to you. 🙏 ❤️

**Pull request merged**